esx_service/vmodl/VsanDockerPersistentVolumeSystem.py

```diff
@@ -0,0 +1,389 @@
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Indicates total storage usage allowed on this datastore
+   """
+   )
+   @Attribute(parent=_name, typ="string")
+   def usage_quota(self):
+       pass
+
+class VsanDockerPersistentVolumeTenant:
+   _name = "vim.vsan.VsanDockerPersistentVolumeTenant"
+   @JavaDocs(parent=_name, docs =
+   """
+   This class encapsulates the Docker Volume Tenant.
+   """
+   )
+      
+   @DataType(name=_name, version=_VERSION)
+   def __init__(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Tenant uuid
+   """
+   )
+   @Attribute(parent=_name, typ="string")
+   def id(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Tenant name
+   """
+   )
+   @Attribute(parent=_name, typ="string")
+   def name(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Tenant description
+   """
+   )
+   @Attribute(parent=_name, typ="string")
+   def description(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Default datastore name
+   """
+   )
+   @Attribute(parent=_name, typ="string")
+   def default_datastore(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Indicates whether the privilege has mount volume privilege
+   """
+   )
+   @Attribute(parent=_name, typ="vim.vsan.VsanDockerPersistentVolumeDatastoreAccessPrivileges")
+   def default_privileges(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Indicates max volume size allowed on this datastore
+   """
+   )
+   @Attribute(parent=_name, typ="string[]", flags=F_OPTIONAL)
+   def vms(self):
+       pass
+
+   @JavaDocs(parent=_name, docs =
+   """
+   Indicates total storage usage allowed on this datastore
+   """
+   )
+   @Attribute(parent=_name, typ="vim.vsan.VsanDockerPersistentVolumeDatastoreAccessPrivileges[]", flags=F_OPTIONAL)
+   def privileges(self):
+       pass     
+
+class VsanDockerPersistentVolumeSystem:
+   ''' This is the API to Docker Persistent Volumes on VSAN'''
+
+   _name = "vim.host.VsanDockerPersistentVolumeSystem"
+
+   @Internal(parent=_name)
+   @ManagedType(name=_name, version=_VERSION)
+   def __init__(self):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Query the tenant list. This method returns a list of tenants, as strings. 
+   The method is blocking on requires DB IO. 
+   It SHOULD be converted to async task, eventually.
+   """
+   )
+   @Method(parent=_name, wsdlName="GetTenantList")
+   @Return(typ="vim.vsan.VsanDockerPersistentVolumeTenantList")
+   def GetTenantList(self):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Create DatastoreAccessPrivileges object
+   """
+   )
+   @Method(parent=_name, wsdlName="CreateDatastoreAccessPrivileges")
+   @Param(name="datastore", typ="string")
+   @Param(name="create_volumes", typ="boolean")
+   @Param(name="delete_volumes", typ="boolean")
+   @Param(name="mount_volumes", typ="boolean")
+   @Param(name="max_volume_size", typ="string")
```

Comment:
as discussed , we should make it consistent (e.g. Create/Delete right, and Mount/Unmount right)